tests/smoke-tests/src/test/java/org/apache/activemq/artemis/tests/smoke/upgradeTest/CompareUpgradeTest.java

```diff
@@ -149,7 +149,12 @@ private void compareDirectories(String expectedFolder, String upgradeFolder, Str
 
                   String expectedString = expectedIterator.next().replace("Expected", "").trim();
                   String upgradeString = upgradeIterator.next().trim();
-                  Assert.assertEquals("error on line " + line + " at " + upgradeFile, expectedString, upgradeString);
+
+                  if (f.getName().endsWith("xml") && expectedString.trim().startsWith("<!--") && upgradeString.trim().startsWith("<!--")) {
+                     logger.debug("Both {} and {} are comments, which I don't really mind what's going on inside the comment", expectedString, upgradeString);
```

Comment:
does anyone have a strong feeling about doing this?

Reply:
I think it's okay to ignore the comments. Should we check if a line includes other content after the comment?
```
<!-- comment text --> <otherContent>...</otherContent>
```

Reply:
I think the comments of files we are not actually upgrading will change infrequently enough and trivially enough (given they are comments), that it would be better just to update the original thing being compared in this case so it matches the post-update expectations, and leave the test handling as it is rather than trying to make it clever enough to ignore comments but not break other necessary checks. We have each all spent longer already than doing that would have taken.

(We could presumably instead even make the upgrader upgrade this comment-only-changed file so it matched the newly generated expectations file.)

Reply:
@gemmellr it's not that simple to generate the jolokia-access.xml

we would have to parse the host-name and if we are relaxing jolokia or not.

At this point we should just ignore the comment on this file. Going beyond that would take a long time.


I will just update the test to ignore comments on this file only. 


We could deal with that in the future if we have to.

Reply:
I wasnt suggesting generating anything. I was saying, we should change the initial pre-upgrade file, which is also the post-upgrade reference in this case, to just match what the test is already causing to be generated by the new broker...since we dont actually upgrade that file currently, just copy it.